marlowe-playground-client/src/Blockly/Internal.js

```diff
@@ -17,6 +17,41 @@ exports.getElementById_ = function (id) {
 };
 
 exports.createWorkspace_ = function (blockly, workspaceDiv, config) {
+  /* Register extensions */
+  /* Silently clean if already registered */
+  try { blockly.Extensions.register('timeout_validator', function () { }); } catch(err) { }
+  blockly.Extensions.unregister('timeout_validator');
+
+  /* Timeout extension (advanced validation for the timeout field) */
+  blockly.Extensions.register('timeout_validator',
+    function () {
+      var thisBlock = this;
```

Comment:
This looks really good! Just one thing I noticed:

I think you should be able to use [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) to "assign" the scope for the functions below instead of a closure. Usually that's what's recommended by styleguides, but it's really up to personal preference =)

Reply:
Thank you. I didn't know about that syntax. I have changed it to use arrow functions now

Reply:
Hm, actually, it seems it is breaking at some point, maybe the minimifier doesn't understand them, or something like that. I am going to revert them for now